app/uk/gov/hmrc/serviceconfigs/scheduler/SchedulerUtils.scala

```diff
@@ -71,6 +71,24 @@ trait SchedulerUtils extends Logging {
         case None    => logger.debug(s"$label cannot run - lock ${lock.lockId} is taken... skipping update")
       }
     }
+
+  import cats.data._
+  type ScheduledItem[A] = ReaderT[WriterT[Future, List[Throwable], *], Option[Throwable], A]
+
+  def runAllAndFailWithFirstError(k: ScheduledItem[Unit])(implicit ec: ExecutionContext) =
```

Comment:
nitpick: could we have an explicit return type on this method since it's public? also, not sure how much need a test case for this new functionality